app/controllers/works_controller.rb

```diff
@@ -415,5 +415,24 @@ def migrated?
 
       params[:submit] == "Migrate"
     end
+
+    # Returns a hash object that can be serialized into something that DataTables
+    # can consume. The `data` elements includes the work's file list all other
+    # properties are used for displaying different data elements related but not
+    # directly on the DataTable object (e.g. the total file size)
+    def file_list_ajax_response(work)
```

Comment:
Notice how we return the file list plus a bunch of other values that depend on the file list